src/main/java/com/couchbase/intellij/tree/PillowFightDialog.java

```diff
@@ -0,0 +1,1507 @@
+
+        gbcCollapsable.gridx++;
+        collapsiblePanel.add(jsonComboBox, gbcCollapsable);
+
+        gbcCollapsable.gridx--;
+        gbcCollapsable.gridy++;
+        //panel.add(new JLabel("NOOP: "), gbc);
+        collapsiblePanel.add(createLabelWithBalloon("NOOP: ", "Use couchbase NOOP operations when running the workload. This mode ignores population, and all other document operations. Useful as the most lightweight workload."), gbcCollapsable);
+
+        gbcCollapsable.gridx++;
+        collapsiblePanel.add(noopComboBox, gbcCollapsable);
+
+        gbcCollapsable.gridx--;
+        gbcCollapsable.gridy++;
+        //panel.add(new JLabel("Subdoc: "), gbc);
+        collapsiblePanel.add(createLabelWithBalloon("Subdoc: ", "Use couchbase sub-document operations when running the workload. In this mode pillowfight will use Couchbase sub-document operations to perform gets and sets of data. This option must be used with --json"), gbcCollapsable);
+
+        gbcCollapsable.gridx++;
+        collapsiblePanel.add(subdocComboBox, gbcCollapsable);
+
+        gbcCollapsable.gridx--;
+        gbcCollapsable.gridy++;
+        //panel.add(new JLabel("Pathcount: "), gbc);
+        collapsiblePanel.add(createLabelWithBalloon("Pathcount: ", "Specify the number of paths a single sub-document operation should contain. By default, each subdoc operation operates on only a single path within the document. You can specify multiple paths to atomically executed multiple subdoc operations within a single command."), gbcCollapsable);
+
+        gbcCollapsable.gridx++;
+        collapsiblePanel.add(pathcountTextField, gbcCollapsable);
+        pathcountTextField.getEmptyText().setText("Enter a number 0 or greater");
+
+        /*
+        JButton toggleButton = new JButton("Show Advanced Options");
+        toggleButton.addActionListener(e -> {
+            boolean isVisible = collapsiblePanel.isVisible();
+            collapsiblePanel.setVisible(!isVisible);
+            if (isVisible) {
+                toggleButton.setText("Show Advanced Options");
+                gbc.weighty = 0.0;
+            } else {
+                toggleButton.setText("Hide Advanced Options");
+                gbc.weighty = 1.0;
+            }
+            panel.revalidate();
+        });
+
+        panel.add(toggleButton, gbc);
+        */
+        gbc.gridy++;
+        CollapsiblePanel collapsiblePanel1 = new CollapsiblePanel("Advanced Options", collapsiblePanel);
+        panel.add(collapsiblePanel1, gbc);
+
+        gbc.gridx--;
+        gbc.gridy++;
+        gbc.gridwidth = GridBagConstraints.REMAINDER;
+        errorMessage = new JLabel("");
+        errorMessage.setForeground(Color.decode("#FF4444"));
+        panel.add(errorMessage, gbc);
+
+        return panel;
+    }
+
+    @Override
+    protected void doOKAction() {
+        List<String> errors = new ArrayList<>();
```

Comment:
Once you replace all the number fields with JSPinner, please revisit this method and remove all the necessary validation